ideasbox/serveradmin/views.py

```diff
@@ -110,3 +114,81 @@ def backup(request):
     return render(request, 'serveradmin/battery.html',
                   {'batteries': batinfo.batteries()})
+
+
+unset_config_data = """scheme_current=None
+interface_current={interface}
+scheme_active=False
+""".format(interface=interface)
+
+def set_wifilist():
+    f = open('.runningconfig', 'r+')
+    read_data = f.read()
+    if len(read_data) == 0:
+        f.write(unset_config_data)
+    print f.read()
+    from wifi.utils import get_property, set_properties
+    to_bool = {'True' : True, 'False' : False, None : False, 'None' : False}
+    # reset scheme_active property
+    essid = get_property('scheme_current')
+    try:
+        is_essid = to_bool[essid]
+    except KeyError:
+        is_essid = True
+        essid = essid.split('--')[0]
+    if is_essid:
+        options = {'wireless-essid' : essid}
+    else:
+        options = None
+    set_properties(None, None, config=options)
+    is_connected = to_bool[get_property('scheme_active')] or False
+    action = {True : _('Disconnect'), False : _('Connect')}
+    try:
+        wifi = Cell.all(interface, sudo=True)
+        # Florian --> config file with interface name
+        for hotspot in wifi:
+            # find out if connected with
+            id_ = '--'.join([hotspot.ssid, hotspot.address])
+            hotspot.is_active = (get_property('scheme_current') == id_)
+            hotspot.is_connected = (hotspot.is_active and is_connected)
+            hotspot.action = action[hotspot.is_connected]
+    except InterfaceError:
+        wifi = ""
+    return wifi, is_connected
+
+
+@staff_member_required
+def wifi(request):
+    set_wifilist()
+    if request.POST:
+        action = request.POST['action']
+        if action == _('Connect'):
+            # we use the addresses to set unicity of the schemes created
+            address = request.POST['address']
+            ssid = request.POST['ssid']
+            cell_kwargs = {'interface' : interface,
+                           'name' : '--'.join([ssid, address]),
+                           'passkey' : request.POST['key']}
+            cell = Cell.where(cell_kwargs['interface'],
+                              lambda cell: cell.address == address)[0]
+            cell_kwargs['cell'] = cell
+            scheme = Scheme.for_cell(**cell_kwargs)
+            if not Scheme.find(cell_kwargs['interface'], scheme.name):
+                scheme.save()
+            try:
+                scheme.activate()
+            except ConnectionError:
+                # just to verify in terminal
+                print "erreur"
+        else:
+            # disconnect
+            import subprocess
+            # to avoid errors in tests
+            subprocess.call(["/sbin/ifdown", interface])
+    # refresh wifilist
+    wifilist, auth = set_wifilist()
+    # causes the long load time because of the scan
+    return render(request, 'serveradmin/wifi.html',
+                  {'wifiList': wifilist,
+                   'AuthOK' : auth})
```

Comment:
no camel case in python: `auth_ok`
btw, this name is a bit ambiguous: are we saying `is_connected` ?